### ISSUE TITLE
Mocks should be handled by cheatable starknet extension

### ISSUE THREAD

**Issue by MaksymilianDemitraszek** [foundry-rs/starknet-foundry#1634]

**Comment by piotmag769** [comment]
It is actually handled by neither of them, it is just a common function they both use

**Comment by Arcticae** [comment]
@MaksymilianDemitraszek can you see if it's  still doable?

**Comment by MaksymilianDemitraszek** [comment]
Yup imo this is doable we just need to separate a `MockCallExtension` and put it before `BlockifierCallExtension` and `CheatableStarknetExtension`
